osu.Game/Screens/Play/HUDOverlay.cs

```diff
@@ -112,8 +114,14 @@ private void load(OsuConfigManager config, NotificationOverlay notificationOverl
 
             ModDisplay.Current.Value = mods;
 
+            hideHealthbar = config.GetBindable<bool>(OsuSetting.HideHealthBar);
             showHud = config.GetBindable<bool>(OsuSetting.ShowInterface);
-            showHud.ValueChanged += visible => visibilityContainer.FadeTo(visible.NewValue ? 1 : 0, duration);
+            showHud.ValueChanged += visible =>
+            {
+                visibilityContainer.FadeTo(visible.NewValue ? 1 : 0, duration);
+                if (!(hideHealthbar.Value && mods.OfType<ModBlockFail>().Any()))
```

Comment:
why is this done here? isn't just doing it in the mod enough?

Reply:
so you're fine with allowing the player to reshow the health bar with Shift + Alt, making the option to be more like hide health bar if can't fail **on start**

Reply:
No, that's not what I'm saying. You're already doing the hiding in the `ModBlockFail` so this is unnecessary.

Reply:
oh well, you can toggle the visibility of HUD Overlay with Shift + Alt, so i made a check to never reshow the health bar if the `hideHealthbar` option was set, i can remove it if you want

Reply:
It's not necessary. Try for yourself. YOu're hiding a different *level* of container.

Reply:
My bad, i forgot that the health bar visibility is untouched by this, sorry for this